compiler/rustc_session/src/session.rs

```diff
@@ -528,9 +534,13 @@ impl Session {
             Err(ErrorGuaranteed::unchecked_claim_error_was_emitted())
         }
     }
+    #[allow(rustc::untranslatable_diagnostic)]
+    #[allow(rustc::diagnostic_outside_of_impl)]
     pub fn span_warn<S: Into<MultiSpan>>(&self, sp: S, msg: impl Into<DiagnosticMessage>) {
```

Comment:
If you add `#[rustc_lint_diagnostics]` to these functions, how many new errors in the compiler do you get?

Reply:
Ok, so I replaced the `allows` with `#[rustc_lint_diagnostics]` in these functions (also in the `early_error_no_abort`, `early_error`, `early_warn`) and I have now 200 compiler errors 

How should I proceed? Try to migrate them all?

Reply:
> And how should I proceed for these functions that the compiler indicates an error? Should I try to migrate them too?

Check that these aren't the same functions that I've added an annotation to in #100754, and if they aren't, then feel free to - otherwise we're just duplicating work.

Reply:
Hey @davidtwco :)

So, I checked and they are the same functions that you added an annotation in your PR.

However, I noticed two functions in `session.rs` that were not migrated (I guess they are new functions that were pulled when I did my rebase). I've also migrated them and you can find the commit [here](https://github.com/rust-lang/rust/pull/101041/commits/169b19b0f2b391f49e6dd4c9cad80af346ea1b8f). 

If you have the time, could you please check it?

After that, I think we can merge this MR :)